tools/bootnode/bootnode.go

```diff
@@ -35,26 +36,36 @@ import (
 	dhtopts "github.com/libp2p/go-libp2p-kad-dht/opts"
 	ma "github.com/multiformats/go-multiaddr"
 	"github.com/pkg/errors"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
 	"github.com/prysmaticlabs/go-bitfield"
 	"github.com/prysmaticlabs/go-ssz"
 	pb "github.com/prysmaticlabs/prysm/proto/beacon/p2p/v1"
 	"github.com/prysmaticlabs/prysm/shared/logutil"
 	"github.com/prysmaticlabs/prysm/shared/params"
+	"github.com/prysmaticlabs/prysm/shared/runutil"
 	"github.com/prysmaticlabs/prysm/shared/version"
 	"github.com/sirupsen/logrus"
 	_ "go.uber.org/automaxprocs"
 )
 
 var (
-	debug        = flag.Bool("debug", false, "Enable debug logging")
-	logFileName  = flag.String("log-file", "", "Specify log filename, relative or absolute")
-	privateKey   = flag.String("private", "", "Private key to use for peer ID")
-	discv5port   = flag.Int("discv5-port", 4000, "Port to listen for discv5 connections")
-	kademliaPort = flag.Int("kad-port", 4500, "Port to listen for connections to kad DHT")
-	metricsPort  = flag.Int("metrics-port", 5000, "Port to listen for connections")
-	externalIP   = flag.String("external-ip", "127.0.0.1", "External IP for the bootnode")
-
-	log = logrus.WithField("prefix", "bootnode")
+	debug         = flag.Bool("debug", false, "Enable debug logging")
+	logFileName   = flag.String("log-file", "", "Specify log filename, relative or absolute")
+	privateKey    = flag.String("private", "", "Private key to use for peer ID")
+	discv5port    = flag.Int("discv5-port", 4000, "Port to listen for discv5 connections")
+	kademliaPort  = flag.Int("kad-port", 4500, "Port to listen for connections to kad DHT")
+	metricsPort   = flag.Int("metrics-port", 5000, "Port to listen for connections")
+	externalIP    = flag.String("external-ip", "127.0.0.1", "External IP for the bootnode")
+	log           = logrus.WithField("prefix", "bootnode")
+	kadPeersCount = promauto.NewGauge(prometheus.GaugeOpts{
+		Name: "bootstrap_node_kaddht_peers",
+		Help: "The current number of kaddht peers of the bootstrap node",
+	})
+	discv5PeersCount = promauto.NewGauge(prometheus.GaugeOpts{
```

Comment:
Want to use NewGaugeFunc? 

https://godoc.org/github.com/prometheus/client_golang/prometheus#NewGaugeFunc

With that, you could remove that RunEvery routine and prometheus would simply run the function when the metrics page is scraped

Reply:
It would end up being more code added cause I would have to call update metrics twice for each of them in the callback, also I wouldn't be able to initialize it at the top as a package var

Reply:
Yeah, i was afraid of that